### PR TITLE
Introducing GroupPortMapper for server to server disruption testing.

### PR DESCRIPTION
**Pull request by mobasherul**

### CODE REVIEW

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/GroupPortMapper.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.dynamic_config.server.configuration.startup;
+
+import org.terracotta.dynamic_config.api.model.Node;
+
+public interface GroupPortMapper {
```
Comment (mathieucarbou) [inline dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/GroupPortMapper.java:20]: service class needs to be in server-api module

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/StartupConfiguration.java [hunk 2]
```diff
@@ -80,6 +84,8 @@
     this.pathResolver = requireNonNull(pathResolver);
     this.substitutor = requireNonNull(substitutor);
     this.objectMapper = objectMapperFactory.create();
+    this.groupPortMapper = ManagedServiceLoader.loadServices(GroupPortMapper.class, GroupPortMapper.class.getClassLoader())
+        .stream().findFirst().get();
```
Comment (mathieucarbou) [inline dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/StartupConfiguration.java:88]: Why not re-using the already existing Sservice Classloader that is inside this class (`classLoader `) ? 
To add you service, this is in `DynamicConfigConfigurationProvider`:
```
  private ClassLoader getServiceClassLoader() {
    return ServerEnv.getServer().getServiceClassLoader(getClass().getClassLoader(),
        DynamicConfigExtension.class,
        LicenseService.class);
  }
```

File: dynamic-config/testing/entity/src/main/java/org/terracotta/dynamic_config/test_support/mapper/TestGroupPortMapper.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.dynamic_config.test_support.mapper;
+
+import org.terracotta.dynamic_config.api.model.Node;
+import org.terracotta.dynamic_config.server.configuration.startup.GroupPortMapper;
+
+public class TestGroupPortMapper implements GroupPortMapper {
```
Comment (mathieucarbou) [inline dynamic-config/testing/entity/src/main/java/org/terracotta/dynamic_config/test_support/mapper/TestGroupPortMapper.java:21]: Why this class is not referenced in a META-INF/services file ?

Comment (mathieucarbou) [inline dynamic-config/testing/entity/src/main/java/org/terracotta/dynamic_config/test_support/mapper/TestGroupPortMapper.java:21]: Why there is no `@OverrideServicd` annotation ?

Note: if core does not provide the ability to override a service, then one solution could be, when you are searching for the alternate implementation to do something like:

```
ManagedServiceLoader.loadServices(GroupPortMapper.class, classloader).stream().findFirst().orElse(new DefaultGroupPortMapperImpl());
```

and then remove the `META-INF/services` file from the `configuration-provider` module and only keep it in your test server plugin.

Comment (mobasherul) [reply dynamic-config/testing/entity/src/main/java/org/terracotta/dynamic_config/test_support/mapper/TestGroupPortMapper.java:21]: missed

File: dynamic-config/server/server-api/src/main/java/org/terracotta/dynamic_config/server/api/GroupPortMapper.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.dynamic_config.server.api;
+
+import org.terracotta.dynamic_config.api.model.Node;
+
+public interface GroupPortMapper {
+  int getGroupPort(Node currentNode, Node node);
```
Comment (ramsai1729) [inline dynamic-config/server/server-api/src/main/java/org/terracotta/dynamic_config/server/api/GroupPortMapper.java:21]: `int getPeerGroupPort(Node peerNode, Node currentNode)` better maybe?

File: dynamic-config/testing/entity/pom.xml [hunk 5]
```diff
@@ -64,5 +64,11 @@
       <artifactId>standard-cluster-services</artifactId>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>org.terracotta.dynamic-config.server</groupId>
+      <artifactId>dynamic-config-configuration-provider</artifactId>
```
Comment (ramsai1729) [inline dynamic-config/testing/entity/pom.xml:69]: still need this dependency?

File: dynamic-config/testing/entity/src/main/java/org/terracotta/dynamic_config/test_support/mapper/TestGroupPortMapper.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.dynamic_config.test_support.mapper;
+
+import org.terracotta.dynamic_config.api.model.Node;
+import org.terracotta.dynamic_config.server.api.GroupPortMapper;
+
+public class TestGroupPortMapper implements GroupPortMapper {
+  private static final String key = "proxy-group-port";
```
Comment (ramsai1729) [inline dynamic-config/testing/entity/src/main/java/org/terracotta/dynamic_config/test_support/mapper/TestGroupPortMapper.java:22]: test-proxy-group-port

File: dynamic-config/testing/entity/pom.xml (hunk 5 above)
Comment (mobasherul) [reply dynamic-config/testing/entity/pom.xml:69]: not needed

**Pull request merged**